### ISSUE TITLE
Variable Distance Buffer should be removed from QGIS 

### ISSUE THREAD

**Issue by antoniolocandro** [qgis/QGIS#46902]
### Feature description

If you go to the docs it mentions the existence of variable distance buffer 
https://docs.qgis.org/testing/en/docs/user_manual/processing_algs/qgis/modelertools.html#variable-distance-buffer 

I open processing and it is not present as I guess the functionality is not present in the normal buffer algorithm, however the algorithm is still reachable using the Python console

![image](https://user-images.githubusercontent.com/4627073/150224403-3f3a4238-54de-4d61-bbd2-5e08b38e2b81.png)

![image](https://user-images.githubusercontent.com/4627073/150225152-8d1a64a8-08c7-4985-89bd-4c5234cf8892.png)

### Additional context

Probably would be a good idea to not have deprecated algorithms present at least in the documentation if it is confirmed it should no longer be used as this would make new people come to the docs and find the methods.

**Comment by DelazJ** [comment]
This is because the buffer variable is in the **graphical modeler**, not in the toolbox (get a closer look to the docs URL). It's not really a deprecated alg.
> Probably would be a good idea to not have deprecated algorithms present at least in the documentation

I think it's more the other way around: the docs should not mention things that are not in the application. This (code) repository is the master, docs has to follow it, not the reverse.

**Comment by antoniolocandro** [comment]
sorry, I failed to read the note 'These tools are only available in the Graphical Modeler. They are not available in the Processing Toolbox', wondering if this should be the first line after Modeler tools to increase visibility and add a warning icon besides it?

Also just as general knowledge what does this implementation provides that the buffer one doesn't except providing a direct shortcut to put in a field from what I can see

**Comment by DelazJ** [comment]
> wondering if this should be the first line after Modeler tools to increase visibility and add a warning icon besides it?

👍 
> Also just as general knowledge what does this implementation provides that the buffer one doesn't except providing a direct shortcut to put in a field from what I can see

No idea. Indeed I think using the DD button one can achieve the same thing but... If a dev can explain...

**Comment by nyalldawson** [comment]
> No idea. Indeed I think using the DD button one can achieve the same thing but... If a dev can explain...


That's correct. The old variable distance buffer algorithm IS deprecated, but it's left visible in models only as we can't remove it entirely and run the risk of breaking a user's model.

IMO it can safely be removed from the documentation, or at least flagged as deprecated in the docs.
